File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/common/fetchoptions/FetchOptions.js
define([ "stjs", "dto/common/fetchoptions/CacheMode" ], function(stjs, CacheMode) {
	var FetchOptions = function() {
		this._count = null;
		this._from = null;
		this._cacheMode = CacheMode.NO_CACHE;
	};
	stjs.extend(FetchOptions, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dto.common.fetchoptions.FetchOptions';
		constructor.serialVersionUID = 1;
		prototype.count = function(count) {
			this._count = count;
			return this;
		};
		prototype.getCount = function() {
			return this._count;
		};
		prototype.from = function(from) {
			this._from = from;
			return this;
		};
		prototype.getFrom = function() {
			return this._from;
		};
		prototype.cacheMode = function(cacheMode) {
			this._cacheMode = cacheMode;
			return this;
		};
		prototype.getCacheMode = function() {
			return this._cacheMode;
		}
	}, {});
	return FetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/common/fetchoptions/FetchOptions.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions;

import java.io.Serializable;
import java.util.HashSet;

import com.fasterxml.jackson.annotation.JsonIgnore;

import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.common.fetchoptions.FetchOptions")
public abstract class FetchOptions<OBJECT> implements Serializable
{

    private static final long serialVersionUID = 1L;

    private Integer count;

    private Integer from;

    private CacheMode cacheMode = CacheMode.NO_CACHE;

    @SuppressWarnings("hiding")
    public FetchOptions<OBJECT> count(Integer count)
    {
        this.count = count;
        return this;
    }

    public Integer getCount()
    {
        return count;
    }

    @SuppressWarnings("hiding")
    public FetchOptions<OBJECT> from(Integer from)
    {
        this.from = from;
        return this;
    }

    public Integer getFrom()
    {
        return from;
    }

    /**
     *
     * @return
     * @deprecated caching modes are ignored
     */
    @Deprecated
    public FetchOptions<OBJECT> cacheMode(CacheMode mode)
    {
        if (mode == null)
        {
            this.cacheMode = CacheMode.NO_CACHE;
        } else
        {
            this.cacheMode = mode;
        }
        return this;
    }

    public CacheMode getCacheMode()
    {
        return cacheMode;
    }

    public abstract SortOptions<OBJECT> sortBy();

    public abstract SortOptions<OBJECT> getSortBy();

    @JsonIgnore
    protected abstract FetchOptionsToStringBuilder getFetchOptionsStringBuilder();

    @Override
    public String toString()
    {
        return getFetchOptionsStringBuilder().toString("", new HashSet<FetchOptions<?>>());
    }
}
